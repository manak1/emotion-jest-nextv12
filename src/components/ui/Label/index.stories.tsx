import { ComponentStory, ComponentMeta } from "@storybook/react"

import Label from "./index"

// More on default export: https://storybook.js.org/docs/react/writing-stories/introduction#default-export
export default {
  title: "Label",
  component: Label,
  argTypes: {
    label: { control: "text", defaultValue: "ラベル" },
  },
} as ComponentMeta<typeof Label>

// More on component templates: https://storybook.js.org/docs/react/writing-stories/introduction#using-args
const Template: ComponentStory<typeof Label> = (args) => {
  return <Label {...args} />
}

export const Default = Template.bind({})
// More on args: https://storybook.js.org/docs/react/writing-stories/args
export const Required = Template.bind({})
Required.args = {
  required: true,
}
