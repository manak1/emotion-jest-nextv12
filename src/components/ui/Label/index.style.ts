import { css } from "@emotion/react"
import { colors, fontSizes } from "~/styles/variables"

export const styles = {
  label: css({
    fontSize: fontSizes.small,
  }),
  required: css({
    color: colors.error,
  }),
}
