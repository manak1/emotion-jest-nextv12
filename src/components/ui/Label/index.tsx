import React from "react"
import { styles } from "./index.style"

type Props = React.ComponentProps<"label"> & {
  label: string
  required?: boolean
}

const Label: React.VFC<Props> = (props) => {
  const { label, required, ...rest } = props
  return (
    <label {...rest} css={styles.label}>
      {required && <span css={styles.required}>*</span>}
      {label}
    </label>
  )
}

export default Label
