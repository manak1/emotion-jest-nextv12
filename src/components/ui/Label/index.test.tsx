import { render } from "~/test/test-utils"
import Label from "./index"

describe("InputTextのテスト", () => {
  it("snapshot", () => {
    const { asFragment } = render(<Label label="ラベル" />)
    expect(asFragment()).toMatchSnapshot()
  })
})
