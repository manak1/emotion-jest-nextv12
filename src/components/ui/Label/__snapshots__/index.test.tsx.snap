// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InputTextのテスト snapshot 1`] = `
<DocumentFragment>
  <label
    class="css-1lu35ut"
  >
    ラベル
  </label>
</DocumentFragment>
`;
